hack/.ci/run-e2e-gke.sh

```diff
@@ -11,12 +11,36 @@ if [ -z "${ARTIFACTS+x}" ]; then
   exit 2
 fi
 
+if [ -z ${KUBECONFIG_DIR+x} ]; then
+  kubeconfigs=("${KUBECONFIG}")
+else
+  kubeconfigs=()
+  for f in $( find "$(realpath "${KUBECONFIG_DIR}")" -maxdepth 1 -type f -name '*.kubeconfig' ); do
+    kubeconfigs+=("${f}")
+  done
+fi
+
 source "$( dirname "${BASH_SOURCE[0]}" )/../lib/kube.sh"
 source "$( dirname "${BASH_SOURCE[0]}" )/lib/e2e.sh"
 parent_dir="$( dirname "${BASH_SOURCE[0]}" )"
 
-trap gather-artifact-on-exit EXIT
+function on-exit {
+  for i in "${!kubeconfigs[@]}"; do
+    KUBECONFIG="${kubeconfigs[$i]}" gather-artifacts "${ARTIFACTS}/must-gather/${i}" &
+    gather_artifacts_bg_pids["${i}"]=$!
+  done
 
+  gather_artifacts_failed=0
+  for pid in "${gather_artifacts_bg_pids[*]}"; do
+    wait "${pid}" || gather_artifacts_failed=1
+  done
+
+  if [ ${gather_artifacts_failed} -eq 1 ]; then
+    echo "Error gathering artifacts" > /dev/stderr
+  fi
+}
```

Comment:
> this is not specific to the GKE so it should be reusable in the lib/e2e

it's not, but having multiple kubeconfigs is specific to this script at this point. Are you saying we should have a single sourced function handling both KUBECONFIG and KUBECONFIG_DIR?

Reply:
> but having multiple kubeconfigs is specific to this script at this point

https://github.com/scylladb/scylla-operator/pull/1837 will need it as well

the future EKS runner would need it too

> Are you saying we should have a single sourced function handling both KUBECONFIG and KUBECONFIG_DIR?

yes, we need a new a function that can collect artifacts in multiple clusters. Unfortunately, I don't think bash is good with function pointer so you likely need dedicated versions of those function instead of a generic runner.

Reply:
done